src/vis/bar/utils.ts

```diff
@@ -0,0 +1,240 @@
+import {merge} from 'lodash';
+import {CategoricalColumn, ColumnInfo, EColumnTypes, ESupportedPlotlyVis, IVisConfig, NumericalColumn, Scales} from '../interfaces';
+import {PlotlyInfo, PlotlyData} from '../interfaces';
+import {getCol} from '../sidebar/utils';
+
+export enum EBarDisplayType {
+    DEFAULT = 'Default',
+    NORMALIZED = 'Normalized',
+}
+
+export enum EBarDirection {
+    VERTICAL = 'Vertical',
+    HORIZONTAL = 'Horizontal',
+}
+
+export enum EViolinOverlay {
+    NONE = 'None',
+    STRIP = 'Strip',
+    BOX = 'Box'
+}
+
+export enum EBarGroupingType {
+    STACK = 'Stacked',
+    GROUP = 'Grouped',
+}
+
+export function isBar(s: IVisConfig): s is IBarConfig {
+    return s.type === ESupportedPlotlyVis.BAR;
+}
+
+export interface IBarConfig {
+    type: ESupportedPlotlyVis.BAR;
+    multiples: ColumnInfo | null;
+    group: ColumnInfo | null;
+    direction: EBarDirection;
+    display: EBarDisplayType;
+    groupType: EBarGroupingType;
+    numColumnsSelected: ColumnInfo[];
+    catColumnsSelected: ColumnInfo[];
+}
+
+const defaultConfig: IBarConfig = {
+    type: ESupportedPlotlyVis.BAR,
+    numColumnsSelected: [],
+    catColumnsSelected: [],
+    group: null,
+    groupType: EBarGroupingType.STACK,
+    multiples: null,
+    display: EBarDisplayType.DEFAULT,
+    direction: EBarDirection.VERTICAL,
+};
+
+export function barMergeDefaultConfig(
+    columns: (NumericalColumn | CategoricalColumn)[],
+    config: IBarConfig
+): IVisConfig {
+    const merged = merge({}, defaultConfig, config);
+
+    const catCols = columns.filter((c) => c.type === EColumnTypes.CATEGORICAL);
+
+    if(merged.catColumnsSelected.length === 0 && catCols.length > 0) {
+        merged.catColumnsSelected.push(catCols[catCols.length - 1].info);
+    }
+
+    return merged;
+}
+
+export function createBarTraces(
+    columns: (NumericalColumn | CategoricalColumn)[],
+    config: IBarConfig,
+    scales: Scales,
+): PlotlyInfo {
+    let counter = 1;
```

Comment:
What does this count? Instances? Plots? Bars? It's unclear without reading the whole code.
Please rename to something more specific.

Reply:
changed to plotCounter in all files